lib/msal-browser/src/app/PublicClientApplication.ts

```diff
@@ -69,6 +70,85 @@ export class PublicClientApplication extends ClientApplication implements IPubli
         return this.acquireTokenPopup(request || DEFAULT_REQUEST);
     }
 
+    /**
+     * This function uses a hidden iframe to fetch an authorization code from the eSTS. There are cases where this may not work:
+     * - Any browser using a form of Intelligent Tracking Prevention
+     * - If there is not an established session with the service
+     *
+     * In these cases, the request must be done inside a popup or full frame redirect.
+     *
+     * For the cases where interaction is required, you cannot send a request with prompt=none.
+     *
+     * If your refresh token has expired, you can use this function to fetch a new set of tokens silently as long as
+     * you session on the server still exists.
+     * @param {@link AuthorizationUrlRequest}
+     *
+     * @returns {Promise.<AuthenticationResult>} - a promise that is fulfilled when this function has completed, or rejected if an error was raised. Returns the {@link AuthResponse} object
+     */
+    async ssoSilent(request: SsoSilentRequest): Promise<AuthenticationResult> {
+        this.preflightBrowserEnvironmentCheck(InteractionType.Silent);
+        this.emitEvent(EventType.SSO_SILENT_START, InteractionType.Silent, request);
+
+        try {
+            // Build cached account by loginHint, sid, or the request itself
+            const accountByLoginHint = request.loginHint && this.getAccountByUsername(request.loginHint);
+            
+            const accountBySid = request.sid && this.getAccountBySessionId(request.sid);
+            if (request.sid && !accountBySid) {
+                this.logger.warning("ssoSilent - sid provided to ssoSilent, but no cached account found. If this is unexpected, confirm sid is enabled as an optional id token claim for your application");
+            }
+
+            const account = request.account || accountByLoginHint || accountBySid;
+
+            // Only checked for cached credentials if forceRefresh is off and there is an account for the request
+            if (!request.forceRefresh && account) {
```

Comment:
forceRefresh could also mean acquire using a refresh token, right? Might not need to check it here as `acquireTokenSilent` should handle that logic

Reply:
I figured it would be best to avoid `acquireTokenSilent` completely if `forceRefresh` is on, as it would semantically mean "give me a new set of tokens, including a new RT", as `forceRefresh` for `acquireTokenSilent` will still use the existing RT.

Reply:
The `forceRefresh` across apps mean only `AT` refresh and not necessarily `RT` considering long lived RT support for all other MSALs. I would leave up to you to interpret that in the way you see fit for `ssoSilent` but this may cause confusion for users when used against both these APIs. (`ssoSilent` and `acquireTokenSilent`) differently.

Reply:
Is there a precedent for a flag to force an RT refresh?

Reply:
I could see an argument for a different flag if we wanted to support the ability to use the cached RT but want to force getting a new AT (which isn't possible, with how I've designed it).

Reply:
.NET has `forceLogin` (which is for setting `prompt=login`): https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/search?q=forcelogin

Would that be acceptable here?

Reply:
`skipCache` could also work.

Reply:
A bit literal, but arguably less confusing/overloaded than `forceRefresh` or `forceLogin`.

Reply:
Decided to go with `skipCache`, defaulting to `true`.